### DISCUSSION TITLE
Include `Metable` trait to all menu related classes

### DISCUSSION THREAD

**Discussion by milewski** [laravel/nova-issues#5960] [Ideas & Feature Requests]
Hi, can we have the `Metable` trait included in the `MenuSection`, `MenuItem`, `MenuGroup`, and `MenuList` since they all include the `Macroable` trait? It doesn't serve much purpose if we can't append extra data before JSON serialization.

For instance, I'm currently working on enhancing the UX of larger menus, in this package: [Collapsible Resource Manager](https://github.com/dcasia/collapsible-resource-manager). It would be far more convenient if I could implement these improvements using the original classes rather than having to extend and create my own.

**Comment by scramatte** [comment]
Hi,

+1 here !

I'm using  @milewski  [dcasia/collapsible-resource-manager](https://github.com/dcasia/collapsible-resource-manager) for  my Nova's apps. This package improve a lot the navigation, overall if application has many modules. 

Mainly I would like to:
- add Icon to MenuGroup,  
- have ability to prepend or append html to menu section
![imagen](https://github.com/laravel/nova-issues/assets/188766/94b7510d-8e68-4a37-af4b-aa48fa54adda)

- be able to link section to default resource including if the section has sub menu (this is not allowed currently)

Thank you
